### PR TITLE
Update QUICHE to e8ddc3873182355137862b4d6417add2b2b8a31d

### PR DESCRIPTION
**Pull request by DavidSchinazi**
Commit Message: Update QUICHE to e8ddc3873182355137862b4d6417add2b2b8a31d
Additional Description: None
Risk Level: Low
Testing: Local
Docs Changes: None
Release Notes: None
Platform Specific Features: None

### CODE REVIEW

**Review by danzh2010 — APPROVED** [review]
LGTM, thanks for the update!

**Comment by davinci26** [comment]
@DavidSchinazi small nit: do you mind updating the title to indicate the direction of the roll so its easier to understand from the commit log.

I.e. update QUICE to `<hash>` or rollback QUICE to `<hash>`

**Comment by DavidSchinazi** [comment]
Done

File: source/common/quic/envoy_quic_proof_source.cc [hunk 1]
```diff
@@ -16,7 +16,10 @@ namespace Quic {
 quic::QuicReferenceCountedPointer<quic::ProofSource::Chain>
 EnvoyQuicProofSource::GetCertChain(const quic::QuicSocketAddress& server_address,
                                    const quic::QuicSocketAddress& client_address,
-                                   const std::string& hostname) {
+                                   const std::string& hostname, bool* cert_matched_sni) {
+  // TODO(DavidSchinazi) parse the certificate to correctly fill in |cert_matched_sni|.
+  *cert_matched_sni = false;
```
Comment (ggreenway) [inline source/common/quic/envoy_quic_proof_source.cc:21]: What's the impact of setting this to false?

**Review by ggreenway — COMMENTED** [review]
Is there anything specific we're picking up in this update, or just keeping up-to-date?

File: source/common/quic/envoy_quic_proof_source.cc (hunk 1 above)
Comment (DavidSchinazi) [reply source/common/quic/envoy_quic_proof_source.cc:21]: There isn't any, the core library simply plumbs this value through (another user needed it) and Envoy doesn't read it on the other side so this is a no-op

**Review by DavidSchinazi — COMMENTED** [review]
This is just us keeping up with upstream QUICHE, no particularly interesting features to report.

**Review by ggreenway — APPROVED** [review]

**Pull request merged**